src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -83,28 +89,58 @@ static MsQuicApi()
         try
         {
             // Check version
-            int arraySize = 4;
+            int arraySize;
+            uint paramSize;
+            int status;
+
+            arraySize = 4;
+            paramSize = (uint)arraySize * sizeof(uint);
             uint* libVersion = stackalloc uint[arraySize];
-            uint size = (uint)arraySize * sizeof(uint);
-            if (StatusFailed(apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_VERSION, &size, libVersion)))
+            status = apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_VERSION, &paramSize, libVersion);
+            if (StatusFailed(status))
             {
+                if (NetEventSource.Log.IsEnabled())
+                {
+                    NetEventSource.Error(null, $"Cannot retrieve {nameof(QUIC_PARAM_GLOBAL_LIBRARY_VERSION)} from MsQuic library: '{status}'.");
+                }
                 return;
             }
+            Version version = new Version((int)libVersion[0], (int)libVersion[1], (int)libVersion[2], (int)libVersion[3]);
 
-            var version = new Version((int)libVersion[0], (int)libVersion[1], (int)libVersion[2], (int)libVersion[3]);
-            if (version < MsQuicVersion)
+            arraySize = 64;
+            paramSize = (uint)arraySize * sizeof(sbyte);
+            sbyte* libGitHash = stackalloc sbyte[arraySize];
+            status = apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH, &paramSize, libGitHash);
+            if (StatusFailed(status))
             {
                 if (NetEventSource.Log.IsEnabled())
                 {
-                    NetEventSource.Info(null, $"Incompatible MsQuic library version '{version}', expecting '{MsQuicVersion}'");
+                    NetEventSource.Error(null, $"Cannot retrieve {nameof(QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH)} from MsQuic library: '{status}'.");
```

Comment:
Did you check it with both official and manually build package?

Reply:
CI is running with the official packages and I run locally with manually built msquic from the its main branch.

Reply:
Note that this won't get hit unless the library is built without this property, which doesn't exists ATM. It's just checking for the status as we do with all msquic calls.